### PR TITLE
refactor-1

### PR DESCRIPTION
**Pull request by sarab96**
refactor of code

### CODE REVIEW

**Review by stsoe — CHANGES_REQUESTED** [review]
These are simple changes for the better.  Please consider my comments.

File: src/xma/src/xmaplugin/xma_utils.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#include "lib/xma_utils.hpp"
+#include "app/xmalogger.h"
+#include "app/xmaerror.h"
+#include "lib/xmalimits_lib.h"
+
+#define XMAUTILS_MOD "xmautils"
+
+namespace xma_core { namespace utils {
+
+int32_t check_xma_session(XmaSession &s_handle) {
```
[the remark this answers is not in the document]
Comment (sarab96) [reply src/xma/src/xmaplugin/xma_utils.cpp]: Agree. Done

File: src/xma/src/xmaplugin/xma_utils.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    XmaHwSessionPrivate *priv1 = (XmaHwSessionPrivate*) s_handle.hw_session.private_do_not_use;
```
[the remark this answers is not in the document]
Comment (sarab96) [reply src/xma/src/xmaplugin/xma_utils.cpp]: Yes agree. Done

File: src/xma/src/xmaplugin/xma_utils.cpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    if (priv1 == nullptr) {
+        xma_logmsg(XMA_ERROR_LOG, XMAUTILS_MOD, "XMASession is corrupted.");
+        return XMA_ERROR;
+    }
+    if (s_handle.session_signature != (void*)(((uint64_t)priv1) | ((uint64_t)priv1->reserved))) {
```
[the remark this answers is not in the document]
Comment (sarab96) [reply src/xma/src/xmaplugin/xma_utils.cpp]: Agreed. Done

**Review by stsoe — APPROVED** [review]

**Pull request merged**